cmd/maya-exporter/app/collector/mock/mock.go

```diff
@@ -0,0 +1,43 @@
+package mock
+
+import (
+	"io/ioutil"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+	"time"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+)
+
+// PrometheusHandler is used to mock the behaviour of prometheus's client-go
+// apis for testing purpose.
+// Here, it registers the collector and then starts a test http server and send
+// a get request to which server reply with the response as expected
+func PrometheusHandler(t *testing.T, col prometheus.Collector) ([]byte, *httptest.Server) {
+	if err := prometheus.Register(col); err != nil {
+		t.Fatalf("collector failed to register: %s", err)
+	}
+
+	server := httptest.NewServer(promhttp.Handler())
+
+	client := http.DefaultClient
+	client.Timeout = 5 * time.Second
+	resp, err := client.Get(server.URL)
+	if err != nil {
+		t.Fatalf("unexpected failed response from prometheus: %s", err)
+	}
+	defer resp.Body.Close()
+
+	buf, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		t.Fatalf("failed reading server response: %s", err)
+	}
+	return buf, server
```

Comment:
What is the purpose of returning both `server response` as well as `server`?

Reply:
server response is used for further comparison and server is used to stop the server after testing is done.

Reply:
@utkarshmani1997 should we make use of stop channel as input to this function?
I guess this is typical idiomatic practice.

Reply:
ok